applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_face_load_condition.cpp

```diff
@@ -72,134 +72,65 @@ void UPwNormalFaceLoadCondition<TDim,TNumNodes>::
     }
 }
 
-//----------------------------------------------------------------------------------------
-template< >
-void UPwNormalFaceLoadCondition<2,2>::
-    InitializeConditionVariables(NormalFaceLoadVariables& rVariables,
-                                 const GeometryType& rGeom)
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void UPwNormalFaceLoadCondition<TDim, TNumNodes>::InitializeConditionVariables(
+    NormalFaceLoadVariables& rVariables,
+    const GeometryType& rGeom)
 {
-    for (unsigned int i=0; i<2; ++i) {
-        rVariables.NormalStressVector[i]     = rGeom[i].FastGetSolutionStepValue(NORMAL_CONTACT_STRESS);
-        rVariables.TangentialStressVector[i] = rGeom[i].FastGetSolutionStepValue(TANGENTIAL_CONTACT_STRESS);
-    }
-}
-
-//----------------------------------------------------------------------------------------
-template< >
-void UPwNormalFaceLoadCondition<3,3>::
-    InitializeConditionVariables(NormalFaceLoadVariables& rVariables,
-                                 const GeometryType& rGeom)
-{
-    for (unsigned int i=0; i<3; ++i) {
+    for (unsigned int i = 0; i < TNumNodes; ++i) {
         rVariables.NormalStressVector[i] = rGeom[i].FastGetSolutionStepValue(NORMAL_CONTACT_STRESS);
     }
-}
-
-//----------------------------------------------------------------------------------------
-template< >
-void UPwNormalFaceLoadCondition<3,4>::
-    InitializeConditionVariables(NormalFaceLoadVariables& rVariables,
-                                 const GeometryType& rGeom)
-{
-    for (unsigned int i=0; i<4; ++i) {
-        rVariables.NormalStressVector[i] = rGeom[i].FastGetSolutionStepValue(NORMAL_CONTACT_STRESS);
+    //
+    if (TDim == 2) {
```

Comment:
This can be constexpr, such that it's checked at compile time
```suggestion
    constexpr if (TDim == 2) {
```